### ISSUE TITLE
Can't use PEP 695 aliases with (legacy) Annotated aliases

### ISSUE THREAD

**Issue by brentyi** [microsoft/pyright#9261]
Hello!

I've been using a library that defines some `Annotated` aliases using the original syntax. My code only needs to run in Python 3.12, I was hoping to use the new alias syntax, but have gotten some unexpected errors.

Here's a reproduction:

```python
from typing import Annotated, TypeAlias, TypeVar

T = TypeVar("T")
A = Annotated[T, "some metadata"]

B = A[int]  # ok
C: TypeAlias = A[int]  # ok
type D = A[int]  # Type parameter "T" is not included in the type parameter list for "A" 
```

It works if I replace the `A` definition with (although I can't do this for external libraries):

```python
type A[T] = Annotated[T, "some metadata"]
```

Is this possible to fix?

Thanks!

**Comment by erictraut** [comment]
I agree this is a bug. Thanks for reporting.

It's an easy fix. It will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.386
